### PR TITLE
Update Poisson_Dirichlet_1d_exactBC.py, create poisson.1d.exactBC.rst, and update pinn_forward.rst

### PR DESCRIPTION
**Pull request by ryankrum**

### CODE REVIEW

**Comment by lululxvi** [comment]
@anranjiao

**Comment by anranjiao** [comment]
> @anranjiao Please check.

It's all good. The example is "Poisson equation in 1D with hard boundary conditions" in the documentation.

File: examples/pinn_forward/Poisson_Dirichlet_1d_exactBC.py [hunk 1]
```diff
@@ -2,4 +2,36 @@
 Implementation of the 1D Poisson equation (Section 3.2.1) in paper https://arxiv.org/abs/2111.02801.
```
Comment (lululxvi) [inline examples/pinn_forward/Poisson_Dirichlet_1d_exactBC.py:2]: Add the information of supported backends

File: examples/pinn_forward/Poisson_Dirichlet_1d_exactBC.py [hunk 2]
```diff
@@ -2,4 +2,36 @@
...
 """
 
-# TODO
+import deepxde as dde
+from deepxde.backend import tf
+import numpy as np
```
Comment (lululxvi) [inline examples/pinn_forward/Poisson_Dirichlet_1d_exactBC.py:7]:
```python
import deepxde as dde
import numpy as np
from deepxde.backend import tf
```

File: docs/demos/pinn_forward/poisson.1d.exactBC.rst [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+Finally, we save and plot the best trained result and the loss history of the model.
+
+.. code-block:: python
+
+    dde.saveplot(losshistory, train_state, issave=True, isplot=True)
+    
+Complete code
+--------------
+.. literalinclude:: ../../../examples/pinn_forward/Poisson_Dirichlet_1d_exactBC.py
+  :language: python.
```
Comment (lululxvi) [inline docs/demos/pinn_forward/poisson.1d.exactBC.rst:103]: No "." at the end.

File: examples/pinn_forward/Poisson_Dirichlet_1d_exactBC.py [hunk 4]
```diff
@@ -2,4 +2,37 @@
 Implementation of the 1D Poisson equation (Section 3.2.1) in paper https://arxiv.org/abs/2111.02801.
```
Comment (lululxvi) [inline examples/pinn_forward/Poisson_Dirichlet_1d_exactBC.py:2]: Delete this comment.

File: examples/pinn_forward/Poisson_Dirichlet_1d_exactBC.py [hunk 5]
```diff
@@ -2,4 +2,37 @@
...
 """
 
-# TODO
+"""Backend supported: tensorflow.compat.v1, tensorflow, pytorch, paddle"""
```
Comment (lululxvi) [inline examples/pinn_forward/Poisson_Dirichlet_1d_exactBC.py:5]: Have you tested all these 4 backends?

**Comment by lululxvi** [comment]
Change the file name `poisson.1d.exactBC.rst` to `poisson.dirichlet.1d.exactbc.rst`

File: docs/demos/pinn_forward.rst [hunk 6]
```diff
@@ -38,7 +39,7 @@ Time-dependent PDEs
    pinn_forward/burgers
    pinn_forward/heat
    pinn_forward/diffusion.1d
-   pinn_forward/diffusion.1d.exactBC
+   pinn_forward/diffusion.1d.exactbc
```
Comment (lululxvi) [inline docs/demos/pinn_forward.rst:42]: Why change this one?

Comment (ryankrum) [reply docs/demos/pinn_forward.rst:42]: I assumed that it should match the name of the .rst file. Should I change it back?

Comment (lululxvi) [reply docs/demos/pinn_forward.rst:42]: Your code is "Poisson", but this line you modified is "diffusion".

Comment (ryankrum) [reply docs/demos/pinn_forward.rst:42]: My apologies, I will change it now

File: examples/pinn_forward/Poisson_Dirichlet_1d_exactBC.py (hunk 5 above)
Comment (lululxvi) [reply examples/pinn_forward/Poisson_Dirichlet_1d_exactBC.py]: @ryankrum

**Comment by lululxvi** [comment]
> @ryankrum Change the file name `poisson.1d.exactBC.rst` to `poisson.dirichlet.1d.exactbc.rst`

@ryankrum

File: examples/pinn_forward/Poisson_Dirichlet_1d_exactBC.py (hunk 5 above)
Comment (ryankrum) [reply examples/pinn_forward/Poisson_Dirichlet_1d_exactBC.py]: I tested all of them and removed pytorch and paddle because they did not seem to work due to the learning rate decay.
